src/plugins/discover/public/context_awareness/composable_profile.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { Profile } from './types';
+
+export type PartialProfile = Partial<Profile>;
+
+export type ComposableAccessor<T> = (getPrevious: T) => T;
+
+export type ComposableProfile<TProfile extends PartialProfile = Profile> = {
+  [TKey in keyof TProfile]?: ComposableAccessor<TProfile[TKey]>;
+};
+
+export const getMergedAccessor = <TKey extends keyof Profile>(
```

Comment:
I think it might be worth adding a couple of lines of JSDoc as it is not 100% clear when this utilities do exactly from the code.